draft-barnes-acme.md

```diff
@@ -222,6 +241,39 @@ To revoke a certificate, the client simply sends a revocation request, signed wi
 
 Note that while ACME is defined with enough flexibility to handle different types of identifiers in principle, the primary use case addressed by this document is the case where domain names are used as identifiers.  For example, all of the identifier validation challenges described in Section {identifier-validation-challenges} below address validation of domain names.  The use of ACME for other protocols will require further specification, in order to describe how these identifiers are encoded in the protocol, and what types of validation challenges the server might require.
 
+# Replay protection
+
+In order to prevent an attacker from replaying ACME messages against
+different ACME resources, or at a different time than they were sent,
+every authenticated ACME request body MUST include, in the Protected
+header, two Private Header Parameters {{RFC7515}}: "acmeURL" and
+"acmeNonce". The value of "acmeURL" MUST be equal to the absolute URL
+to which the client is submitting the request. The server MUST reject
+client messages sent to a URL that does not exactly match the value of the
+"acmeURL" parameter in the Protected header.
+
+The value of "acmeNonce" MUST be equal to a nonce previously received
+from the server in the ACME-Nonce HTTP header of any message.
+
+The ACME server MUST send an ACME-Nonce HTTP header in response to any POST
+request from the client. The header MUST be unique in each response,
+with high probability, and consist of a random 16-byte octet string,
+Base64-encoded.
+
```

Comment:
Can we get this represented more formally, the same way as most of the other parameters, like say |type| and |token|